streams/src/test/java/org/apache/kafka/streams/integration/ResetIntegrationTest.java

```diff
@@ -187,7 +185,8 @@ public void testReprocessingFromScratchAfterResetWithIntermediateUserTopic() thr
         streams.close();
 
         assertThat(resultRerun, equalTo(result));
-        assertThat(resultRerun2, equalTo(result2));
+        final int maxToCompare = Math.min(result2.size(), resultRerun2.size());
+        assertThat(resultRerun2.subList(0, maxToCompare), equalTo(result2.subList(0, maxToCompare)));
```

Comment:
The test is somewhat non-deterministic with respect to the expected result. Because, `CLUSTER` mock time get's initialized with `System.currentTimeMillis`, the timestamps assigned to the input topic records vary, and thus, they can fall into different windows in each run. And therefore, we don't know exactly how many records we need to expect. For this reason, we only wait for the first 10 records -- however, `IntegrationTestUtils.waitUntilMinKeyValueRecordsReceived` does not return exactly 10 records, but at-least 10 records, because the actual fetch size of the consumer can vary -- thus, we can end up with a different number of returned record for both runs (even if the topic contains the same data). However, the common prefix must be the same and thus we use it for the comparison.

An alternative (more complex) fix would be, to use the initial mock timestamp to figure out the windows in which records fall into, and thus compute the number the expected result record exactly. But I don't think this would make the test better, but only complicates the code.

Does this make sense? \cc @enothereska @guozhangwang